The job **`x86_64-gnu-tools`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/8653519870/job/23728855431) [(plain)](https://github.com/rust-lang/rust/commit/036e538e06589d84f058e7d90d6172b519e6616a/checks/23728855431/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
   Compiling getopts v0.2.21
error[E0658]: use of unstable library feature 'panic_can_unwind'
   --> library/proc_macro/src/bridge/client.rs:290:62
    |
290 |             if force_show_panics || !is_available() || !info.can_unwind() {
    |
    = note: see issue #92988 <https://github.com/rust-lang/rust/issues/92988> for more information
    = help: add `#![feature(panic_can_unwind)]` to the crate attributes to enable
    = note: this compiler was built on 2024-03-18; consider upgrading it if it is out of date
```

</details>

--- a/library/proc_macro/src/bridge/client.rs
+++ b/library/proc_macro/src/bridge/client.rs
@@ -283,7 +283,11 @@ fn maybe_install_panic_hook(force_show_panics: bool) {
     HIDE_PANICS_DURING_EXPANSION.call_once(|| {
         let prev = panic::take_hook();
         panic::set_hook(Box::new(move |info| {
-            if force_show_panics || !is_available() {
+            // We normally report panics by catching unwinds and passing the payload from the
+            // unwind back to the compiler, but if the panic doesn't unwind we'll abort before
+            // the compiler has a chance to print an error. So we special-case PanicInfo where
+            // can_unwind is false.
+            if force_show_panics || !is_available() || !info.can_unwind() {
                 prev(info)
             }
         }));

--- a/library/proc_macro/src/lib.rs
+++ b/library/proc_macro/src/lib.rs
@@ -30,6 +30,7 @@
 #![feature(maybe_uninit_write_slice)]
 #![feature(negative_impls)]
 #![feature(new_uninit)]
+#![feature(panic_can_unwind)]
 #![feature(restricted_std)]
 #![feature(rustc_attrs)]
 #![feature(min_specialization)]

